compiler/rustc_hir/src/hir.rs

```diff
@@ -2715,6 +2716,17 @@ impl FnRetTy<'_> {
     }
 }
 
+/// Represents `for<...>` binder before a closure
+#[derive(Copy, Clone, Debug, HashStable_Generic)]
+pub enum ClosureBinder<'hir> {
+    /// Binder is not specified.
+    Default,
+    /// Binder is specified.
+    ///
+    /// Span points to the whole `for<...>`.
+    For { hir_id: HirId, span: Span, generic_params: &'hir [GenericParam<'hir>] },
```

Comment:
Where is this `HirId` used?

Reply:
I'm pretty sure only here: https://github.com/rust-lang/rust/blob/67ff515e8b08d0e77bf5576d2393090b02139b8a/compiler/rustc_resolve/src/late.rs#L3546

But since [you say](https://github.com/rust-lang/rust/pull/98705#discussion_r911281486) that it should use `expr.id`, maybe this is not needed.

Reply:
Removed in b30f7b27effda91a018a0222643ba192c5b6a8fe